### DISCUSSION TITLE
Make loading dialogue non-obstructing

### DISCUSSION THREAD

**Discussion by rnkn** [isontheline/pro.webssh.net#882] [Ideas]
Hello,

Although it seems improbable to enjoy using WebSSH more than I do now, I think it may be possible if the loading dialogue was non-obstructing on iOS, i.e. when making a new SSH/SFTP connection, or navigating between paths in SFTP, this dialogue appears:

![IMG_0113](https://user-images.githubusercontent.com/1256849/235340200-e19ffaea-8b7e-4129-acca-a11288037ec3.jpeg)

I lack patience and dislike waiting.

But this dialogue obstructs use of another tab in WebSSH but also other apps, e.g. I cannot cmd-tab. Could this dialogue be replaced with something like the Safari refresh spinner or its blue progress bar? Something out-of-the-way and subtle?

Ideally I could tap a path in SFTP then while it loads I could switch over to a SSH session within WebSSH or another app and do something else. Does that make sense?

**Comment by isontheline** [comment]
Hello Paul @rnkn 👋

Of course it make sense!

This dialog has been implemented years ago to address an issue on iOS but now you're right it should be done otherwise.

I will open an issue about your request.

I can't give you any ETA as WebSSH suffers one major memory pressure trouble on macOS that I need to address shortly.

But your request make sense!
